server/src/test/java/org/elasticsearch/transport/RemoteClusterServiceTests.java

```diff
@@ -81,6 +81,12 @@ public void testSettingsAreRegistered() {
         assertTrue(ClusterSettings.BUILT_IN_CLUSTER_SETTINGS.contains(RemoteConnectionStrategy.REMOTE_CONNECTION_MODE));
         assertTrue(ClusterSettings.BUILT_IN_CLUSTER_SETTINGS.contains(SniffConnectionStrategy.REMOTE_CONNECTIONS_PER_CLUSTER));
         assertTrue(ClusterSettings.BUILT_IN_CLUSTER_SETTINGS.contains(SniffConnectionStrategy.REMOTE_CLUSTER_SEEDS));
+        assertTrue(
+            (ClusterSettings.CCX2_FEATURE_FLAG_ENABLED
+                && ClusterSettings.BUILT_IN_CLUSTER_SETTINGS.contains(SniffConnectionStrategy.REMOTE_CLUSTER_AUTHORIZATION))
+                || (ClusterSettings.CCX2_FEATURE_FLAG_ENABLED == false
+                    && ClusterSettings.BUILT_IN_CLUSTER_SETTINGS.contains(SniffConnectionStrategy.REMOTE_CLUSTER_AUTHORIZATION) == false)
```

Comment:
I think this would be clearer like:
```
if (ClusterSettings.CCX2_FEATURE_FLAG_ENABLED) {
  assertTrue(ClusterSettings.BUILT_IN_CLUSTER_SETTINGS.contains(SniffConnectionStrategy.REMOTE_CLUSTER_AUTHORIZATION))
} else {
  assertFalse(ClusterSettings.BUILT_IN_CLUSTER_SETTINGS.contains(SniffConnectionStrategy.REMOTE_CLUSTER_AUTHORIZATION)
}
```

Reply:
For other feature flags, we ensure the flag is enabled when running most tests. I wonder if we should do the same here.

We could add a minimal test to verify that the setting is not available when the flag is not set. I think we have not done so for for instance tsdb though (@nik9000 may know if this is true).

I'd prefer to keep this similar, since otherwise test code is going to have these checks spread all over as we move forward with this feature. I think the feature is designed such that if `authorization` is not set, it does nothing and we can validate this in tests.

Also, validating it this way means it will pass our regular PR builds etc., but fail in release builds, which is sort of late in the process. If we think we need to validate that the setting is not registered when the feature is not enabled, I'd prefer a separate test for that (which can be done in a few different ways).

Reply:
> We could add a minimal test to verify that the setting is not available when the flag is not set. I think we have not done so for for instance tsdb though (@nik9000 may know if this is true).

We haven't. I should do that though. It's kind of tricky to run it without the feature flags because we auto-enable all of the feature flags on `-SNAPSHOT` builds. So we'd have to perform a "release" build or skip the test. And we'd need a special `qa` style sub-module for it because we tend to enable the feature flag all over the place.

I do love the paranoia of making sure that we don't enable enable the option without the flag. @mark-vieira might be able to ballpark how much effort it'd take to set something like this up. I do like the idea.

Reply:
> Also, validating it this way means it will pass our regular PR builds etc., but fail in release builds, which is sort of late in the process. If we think we need to validate that the setting is not registered when the feature is not enabled, I'd prefer a separate test for that (which can be done in a few different ways).

We run the release builds nightly, just to catch stuff like this. *but* we're not as good at keeping them working. There is a `test-release` flag that runs the release build on PRs though! It just takes hours.

Reply:
We need to figure out what our policy is on feature flags. When I was sorting out issues with BWC testing I found that we have no real convention regarding their implementation (see [this comment](https://github.com/elastic/release-eng/issues/70#issuecomment-1199503493)).

How we handle and test feature flags needs some more discussion, and probably outside the context of this PR. Whatever we decide will likely involve refactoring existing feature flag usages as well. I'm not sure who the right audience is for that discussion though.

Reply:
In order to unblock this PR, I suggest to simply do like tsdb until we have the proper test handling.

We can still add a unittest for a function deriving the contents of `ClusterSettings.BUILT_IN_CLUSTER_SETTINGS`, though we would have to rely on the input to it being correct. I think this is sufficient for now.

Reply:
For now, I updated the test base based the example @gwbrown provided.

I don't have any context on TSDB. If you are able to point me to the example you had in mind, it would help me know what you intended. I can take a look and make adjustments.

Reply:
No update for 2 weeks so I assume the change is OK and I will resolve this comment.